src/clients/tenant-client.js

```diff
@@ -0,0 +1,109 @@
+const { BaseEvernodeClient } = require('./base-evernode-client');
+const { EvernodeEvents, MemoFormats, MemoTypes, ErrorCodes, ErrorReasons } = require('../evernode-common');
+const { EventEmitter } = require('../event-emitter');
+const { EncryptionHelper } = require('../encryption-helper');
+const { XrplAccount } = require('../xrpl-account');
+const { TransactionHelper } = require('../transaction-helper');
+
+const ACQUIRE_WATCH_PREFIX = 'acquire_';
+
+const TenantEvents = {
+    AcquireSuccess: EvernodeEvents.AcquireSuccess,
+    AcquireError: EvernodeEvents.AcquireError,
+}
+
+class TenantClient extends BaseEvernodeClient {
+
+    #respWatcher = new EventEmitter();
+
+    constructor(xrpAddress, xrpSecret, options = {}) {
+        super(xrpAddress, xrpSecret, Object.values(TenantEvents), true, options);
+
+        this.on(TenantEvents.AcquireSuccess, async (ev) => {
+            this.#respWatcher.emit(ACQUIRE_WATCH_PREFIX + ev.acquireRefId, { success: true, data: ev.payload, transaction: ev.transaction });
+        });
+        this.on(TenantEvents.AcquireError, async (ev) => {
+            this.#respWatcher.emit(ACQUIRE_WATCH_PREFIX + ev.acquireRefId, { success: false, data: ev.reason, transaction: ev.transaction });
+        });
+    }
+
+    async prepareAccount() {
+        try {
+            if (!await this.xrplAcc.getMessageKey())
+                await this.xrplAcc.setAccountFields({ MessageKey: this.accKeyPair.publicKey });
+        }
+        catch (err) {
+            console.log("Error in preparing user xrpl account for Evernode.", err);
+        }
+    }
+
+    async acquireLeaseSubmit(hostAddress, requirement, options = {}) {
+        const host = new XrplAccount(hostAddress, null, { xrplApi: this.xrplApi });
+        const hostNfts = (await host.getNfts()).filter(nft => nft.URI.startsWith(TransactionHelper.asciiToHex('evrlease').toUpperCase()));
```

Comment:
There's a constant for `TransactionHelper.asciiToHex('evrlease').toUpperCase()` -> `EvernodeConstants.LEASE_NFT_PREFIX_HEX`